addons/mail/static/src/models/messaging_initializer/messaging_initializer.js

```diff
@@ -302,7 +302,7 @@ function factory(dependencies) {
             }
             if (current_partner) {
                 const partnerData = this.messaging.models['mail.partner'].convertData(current_partner);
-                partnerData.user = insert({ id: currentUserId });
+                partnerData.user = insert({ id: currentUserId, isInternalUser: current_partner.is_internal_user });
```

Comment:
`is_internal_user` should already be handled in partner `convertData`, are you sure we need this diff?

Reply:
Yes you are right `is_internal_user` is handled in partner `convertData` and here we will get `is_internal_user` in partnerData.user but after `partnerData.user = insert({ id: currentUserId });` we will only get `id` in partnerData.user because insertion of `id` only. Thus, we need to insert `is_internal_user` here.